src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -492,196 +543,248 @@ protected void Accepting(object message)
-                .With<Send>(send =>
+                        var handle = assoc.Association.AsInstanceOf<AkkaProtocolHandle>();
+                        var drop = matchesQuarantine(handle);
+                        if (drop)
+                            handle.Disassociate();
+                        return !drop;
+                    }).ToList();
+                    return new KeyValuePair<IActorRef, List<InboundAssociation>>(x.Key, associations);
+                }).ToDictionary(k => k.Key, v => v.Value);
+
+                if (quarantine.Uid.HasValue)
                 {
-                    var recipientAddress = send.Recipient.Path.Address;
-                    Func<int?, IActorRef> createAndRegisterWritingEndpoint = refuseUid => endpoints.RegisterWritableEndpoint(recipientAddress,
-                        CreateEndpoint(recipientAddress, send.Recipient.LocalAddressToUse,
-                            _transportMapping[send.Recipient.LocalAddressToUse], settings, writing: true,
-                            handleOption: null, refuseUid: refuseUid), refuseUid);
-
-                    endpoints.WritableEndpointWithPolicyFor(recipientAddress).Match()
-                        .With<Pass>(
-                            pass =>
-                            {
-                                pass.Endpoint.Tell(send);
-                            })
-                        .With<Gated>(gated =>
+                    _endpoints.MarkAsQuarantined(quarantine.RemoteAddress, quarantine.Uid.Value, Deadline.Now + _settings.QuarantineDuration);
+                    _eventPublisher.NotifyListeners(new QuarantinedEvent(quarantine.RemoteAddress, quarantine.Uid.Value));
+                }
+            });
+
+            Receive<Send>(send =>
+            {
+                var recipientAddress = send.Recipient.Path.Address;
+                Func<int?, IActorRef> createAndRegisterWritingEndpoint = refuseUid => _endpoints.RegisterWritableEndpoint(recipientAddress,
+                    CreateEndpoint(recipientAddress, send.Recipient.LocalAddressToUse,
+                        _transportMapping[send.Recipient.LocalAddressToUse], _settings, writing: true,
+                        handleOption: null, refuseUid: refuseUid), uid: null, refuseUid: refuseUid);
+
+                _endpoints.WritableEndpointWithPolicyFor(recipientAddress).Match()
+                    .With<Pass>(
+                        pass =>
                         {
-                            if(gated.TimeOfRelease.IsOverdue) createAndRegisterWritingEndpoint(null).Tell(send);
-                            else Context.System.DeadLetters.Tell(send);
+                            pass.Endpoint.Tell(send);
                         })
-                        .With<Quarantined>(quarantined =>
-                        {
+                    .With<Gated>(gated =>
+                    {
+                        if (gated.TimeOfRelease.IsOverdue) createAndRegisterWritingEndpoint(null).Tell(send);
```

Comment:
If we're `Gated`:

1. If the `timeOfRelease` `Deadline` is overdue, we'll create a new writing endpoint for this address and send it the message.
2. Otherwise, the message will go to deadletters. Not delivering any messages to the remote system while it's gated and time has not yet expired.